libafl_bolts/src/llmp.rs

```diff
@@ -1213,18 +1222,24 @@ where
         sender_id: ClientId,
         next_min_shmem_size: usize,
     ) -> Result<LlmpSharedMap<<SP>::ShMem>, Error> {
-        if self.unused_shmem_cache.is_empty() {
-            // No cached maps that fit our need, let's allocate a new one.
-            Ok(LlmpSharedMap::new(
-                sender_id,
-                self.shmem_provider.new_shmem(next_min_shmem_size)?,
-            ))
-        } else {
-            // We got cached shmems laying around, hand it out, if they are large enough.
-            let mut cached_shmem = self
-                .unused_shmem_cache
-                .remove(self.unused_shmem_cache.len() - 1);
+        let cached_shmem = self
+            .unused_shmem_cache
+            .iter()
+            .position(|cached_shmem| {
+                let page = &(*shmem2page(&cached_shmem.shmem));
+                let receivers_joined_count = page.receivers_joined_count.load(Ordering::Relaxed);
+                debug_assert_ne!(receivers_joined_count, 0);
```

Comment:
why should this debug assert have receivers_joined_count != 0?
here in this closure we are searching for the *free*-ed pages right?
so we are looking for any page that is
receivers_joined_count = 0 and receivers_left_count = 0. (Am I right?)

Reply:
- A page may never be freed if `receivers_joined_count == 0` - else we would have unmapped a page that was nevermapped (which will break on proper shared pages since the page gets freed)
- We are looking for a page that's `receivers_joined_count > 0` and `receivers_left_count == receivers_joined_count` (so, nobody is actively reading from this page anymore, either).